inginious/agent/docker_agent/__init__.py

```diff
@@ -441,7 +441,10 @@ async def create_student_container(self, parent_info, socket_id, environment_nam
                 return
 
             environment = self._containers[environment_type][environment_name]["id"]
-            runtime = self._containers[environment_type][environment_name]["runtime"]
+            if run_as_root:
+                runtime = "kata-runtime"
+            else:
+                runtime = "io.containerd.runc.v2"
```

Comment:
This won't work, as runtime names are per-instance (its in the docker's config).
The task should be able to indicate the name of the runtime directly, rather than "run_as_root"

Reply:
Well, it works locally but I don't know exactly the differences when running on the deployed version :/ 
Anyway, indeed being able to indicade the runtime name would be more efficient (no bug if the runtime version name change for example) but I wanted a kind of shortcut to tell to start as docker or root without requiring more knowledge... :/ 
I can simply ask the runtime name as argument instead of "run_as_root" but I suppose it may be less easy to use.

Reply:
Indeed, hardcoding the runtimes should be avoided as it was designed to support multiple runtimes from the start.

@GuillaumeDerval I actually remember being able to run containers with different runtimes on the same instance as long as they are available, but maybe I smoked (not sure your first remark is about that).

Reply:
This should be resolved now, just pushed modifications: https://github.com/darkyne/INGInious/blob/496b3dc8faef05d06857b9c1c93d52235964e1e0/inginious/agent/docker_agent/__init__.py#L446

Now when the user manually entered an environment as input, it tries to use it.
If nothing was specified (which is generally the case), if run_as_root then it takes the first available runtime which allows run_as_root (each DockerRuntime object has a run_as_root value) otherwise, if not run_as_root it takes the first available runtime wich has run_as_root set to False.

I hope this solved the problem :)